test/natalie/compiler2/bootstrap_test.rb

```diff
@@ -124,6 +124,40 @@ def test_or
     true || fail
   end
 
+  def test_case
+    foo = :bar
+    case foo
+    when :baz
+      fail
+    when :bar
+      foo = :foo
+    else
+      fail
+    end
+    assert_eq(foo, :foo)
+
+    case foo
+    when :foo, fail
+      foo = :bar
+    when fail
+      # pass
+    else
+      fail
+    end
+    assert_eq(foo, :bar)
+
+    case
+    when false
+      fail
+    when nil
+      fail
+    when 1 + 1 == 2
+      foo = :foo
+    end
+    assert_eq(foo, :foo)
```

Comment:
We should add a test for a `case` that has no `else` and does not match any of the branches:

```sh
bin/natalie -i -e "case 0; when 1; end"
natalie/lib/natalie/vm.rb:49:in `pop': out of stack (RuntimeError)
```

Reply:
Seems to be an error in the interpreter,
The cpp backend seems to be able to compile this...

Reply:
Found it
We should push nil in empty blocks